packages/test/test-end-to-end-tests/src/test/benchmark/DocumentMultipleDds.ts

```diff
@@ -0,0 +1,241 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import {
+	ContainerRuntime,
+	IContainerRuntimeOptions,
+	ISummarizer,
+} from "@fluidframework/container-runtime";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+	ContainerRuntimeFactoryWithDefaultDataStore,
+	DataObject,
+	DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { SharedMap } from "@fluidframework/map";
+import { SharedString } from "@fluidframework/sequence";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { IContainer } from "@fluidframework/container-definitions";
+import {
+	createAndAttachContainer,
+	createLoader,
+	createSummarizerFromFactory,
+	summarizeNow,
+} from "@fluidframework/test-utils";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { IDocumentLoaderAndSummarizer, IDocumentProps, ISummarizeResult } from "./DocumentCreator";
+
+class TestDataObject extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+
+	public get _context() {
+		return this.context;
+	}
+
+	private readonly mapKey = "map";
+	public map!: SharedMap;
+
+	private readonly sharedStringKey = "sharedString";
+	public sharedString!: SharedString;
+
+	protected async initializingFirstTime() {
+		const sharedMap = SharedMap.create(this.runtime);
+		this.root.set(this.mapKey, sharedMap.handle);
+
+		const sharedString = SharedString.create(this.runtime);
+		this.root.set(this.sharedStringKey, sharedString.handle);
+	}
+
+	protected async hasInitialized() {
+		const mapHandle = this.root.get<IFluidHandle<SharedMap>>(this.mapKey);
+		assert(mapHandle !== undefined, "SharedMap not found");
+		this.map = await mapHandle.get();
+
+		const sharedStringHandle = this.root.get<IFluidHandle<SharedString>>(this.sharedStringKey);
+		assert(sharedStringHandle !== undefined, "SharedMatrix not found");
+		this.sharedString = await sharedStringHandle.get();
+	}
+}
+
+const runtimeOptions: IContainerRuntimeOptions = {
+	summaryOptions: {
+		summaryConfigOverrides: {
+			state: "disabled",
+		},
+	},
+};
+
+// implement IDocumentLoader methods
+export class DocumentMultipleDds implements IDocumentLoaderAndSummarizer {
+	private _mainContainer: IContainer | undefined;
+	private containerRuntime: ContainerRuntime | undefined;
+	private mainDataStore: TestDataObject | undefined;
+	private readonly dsCounts: number;
+	private readonly dsCountPerIteration: number;
```

Comment:
This can be a fixed number. You don't want to create too many data stores in one go because it will hit the op limit in ODSP. You can set this to 250 or something for all tests.

Reply:
I could but I want it to be a multiple of the dataStoreCounts